iceoryx2-cal/tests/resizable_shared_memory_tests.rs

```diff
@@ -0,0 +1,954 @@
+        let tr_chunk_2 = unsafe {
+            sut_viewer
+                .register_and_translate_offset(chunk_2.offset)
+                .unwrap()
+        };
+        assert_that!(sut_viewer.number_of_active_segments(), eq 2);
+        let tr_chunk_3 = unsafe {
+            sut_viewer
+                .register_and_translate_offset(chunk_3.offset)
+                .unwrap()
+        };
+        assert_that!(sut_viewer.number_of_active_segments(), eq 3);
+        let tr_chunk_4 = unsafe {
+            sut_viewer
+                .register_and_translate_offset(chunk_4.offset)
+                .unwrap()
+        };
+        assert_that!(sut_viewer.number_of_active_segments(), eq 4);
+
+        assert_that!(unsafe { *(tr_chunk_1 as *mut u8) }, eq value_1);
+        assert_that!(unsafe { *(tr_chunk_2 as *mut u16) }, eq value_2);
+        assert_that!(unsafe { *(tr_chunk_3 as *mut u32) }, eq value_3);
+        assert_that!(unsafe { *(tr_chunk_4 as *mut u64) }, eq value_4);
+    }
+
+    #[test]
+    fn unregister_offset_in_view_releases_unused_segments<
+        Shm: SharedMemory<DefaultAllocator>,
+        Sut: ResizableSharedMemory<DefaultAllocator, Shm>,
+    >() {
+        let config = generate_isolated_config::<Sut>();
+        let storage_name = generate_name();
+
+        let sut = Sut::MemoryBuilder::new(&storage_name)
+            .config(&config)
+            .max_chunk_layout_hint(Layout::new::<u8>())
+            .max_number_of_chunks_hint(1)
+            .allocation_strategy(AllocationStrategy::BestFit)
+            .create()
+            .unwrap();
+
+        let chunk_1 = sut.allocate(Layout::new::<u8>()).unwrap().offset;
+        let chunk_2 = sut.allocate(Layout::new::<u16>()).unwrap().offset;
+        let chunk_3 = sut.allocate(Layout::new::<u32>()).unwrap().offset;
+        let chunk_4 = sut.allocate(Layout::new::<u64>()).unwrap().offset;
+
+        let sut_viewer = Sut::ViewBuilder::new(&storage_name)
+            .config(&config)
+            .open()
+            .unwrap();
+
+        unsafe { sut_viewer.register_and_translate_offset(chunk_1).unwrap() };
+        unsafe { sut_viewer.register_and_translate_offset(chunk_2).unwrap() };
+        unsafe { sut_viewer.register_and_translate_offset(chunk_3).unwrap() };
+        unsafe { sut_viewer.register_and_translate_offset(chunk_4).unwrap() };
+
+        assert_that!(sut_viewer.number_of_active_segments(), eq 4);
+
+        unsafe { sut_viewer.unregister_offset(chunk_1) };
+        assert_that!(sut_viewer.number_of_active_segments(), eq 3);
+        unsafe { sut_viewer.unregister_offset(chunk_2) };
+        assert_that!(sut_viewer.number_of_active_segments(), eq 2);
+        unsafe { sut_viewer.unregister_offset(chunk_3) };
+        assert_that!(sut_viewer.number_of_active_segments(), eq 1);
+        unsafe { sut_viewer.unregister_offset(chunk_4) };
+        assert_that!(sut_viewer.number_of_active_segments(), eq 1);
```

Comment:
So, it remains 1 even after releasing chunk_4 because it's the last added segment and kept for later, right?

Reply:
correct